robot/src/main/java/us/ilite/lib/drivers/SparkMaxFactory.java

```diff
@@ -0,0 +1,75 @@
+/*----------------------------------------------------------------------------*/
+/* Copyright (c) 2018 FIRST. All Rights Reserved.                             */
+/* Open Source Software - may be modified and shared by FRC teams. The code   */
+/* must be accompanied by the FIRST BSD license file in the root directory of */
+/* the project.                                                               */
+/*----------------------------------------------------------------------------*/
+
+package us.ilite.lib.drivers;
+
+import com.revrobotics.CANEncoder;
+import com.revrobotics.CANSparkMax;
+import com.revrobotics.CANSparkMax.ExternalFollower;
+import com.revrobotics.CANSparkMax.IdleMode;
+import com.revrobotics.CANSparkMaxLowLevel.ConfigParameter;
+import com.revrobotics.CANSparkMaxLowLevel.MotorType;
+import com.revrobotics.CANSparkMaxLowLevel.PeriodicFrame;
+
+import us.ilite.common.config.SystemSettings;
+
+/**
+ * This is a factory class for the Spark MAX motor controller that re-configures
+ * all settings to our defaults. Note that settings must be explicitly flashed
+ * to the Spark MAX in order to persisten across power cycles.
+ * 
+ */
+public class SparkMaxFactory {
+
+    public static class Configuration {
+        public int CAN_TIMEOUT = 100;
+        public int CONTROL_FRAME_PERIOD = 5;
+        public IdleMode IDLE_MODE = IdleMode.kBrake;
+        public boolean IS_INVERTED = false;
+        public int STATUS_0_PERIOD_MS = 10;

```

Comment:
Make sure these are the default frame periods; we don't want to auto-overload the CAN bus

Reply:
I set these to the same values as the TalonSRXFactory, would you like me to change it to 10? (That's the default) My train of thought was that the cost of setting the master controller to a higher rate would be offset by setting the slave control frame period to 100ms.

Reply:
I'd set it to the default from Rev, then we can adjust as needed.  If we haven't tested full utilization, there isn't any reason to change from the defaults (except for specific cases of sensors).